### PR TITLE
Dev to start 27 mar

### PR DESCRIPTION
**Pull request by sasurobert**

### CODE REVIEW

**Comment by AdoAdoAdo** [comment]
there seem to be a lot of fixes after merge, which were actually not differences between what has been separately changed on the feature branch and in dev.

Did something go wrong with the merge?

I would suggest to redo the merge.

File: node/heartbeat/monitor.go [hunk 1]
```diff
@@ -198,14 +198,13 @@ func (m *Monitor) loadHbmiFromStorer(pubKey string) (*heartbeatMessageInfo, erro
 	}
 	receivedHbmi.lastUptimeDowntime = crtTime
 	receivedHbmi.genesisTime = m.genesisTime
-	receivedHbmi.peerType = m.computePeerType([]byte(pubKey), receivedHbmi.computedShardID)
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply node/heartbeat/monitor.go]: asked bogdan, and appearantly not. It is not there is dev.

File: go.mod [hunk 2]
```diff
@@ -11,6 +11,7 @@ require (
 	github.com/beevik/ntp v0.2.0
 	github.com/btcsuite/btcd v0.20.1-beta
 	github.com/btcsuite/btcutil v0.0.0-20190425235716-9e5f4b9a998d
+	github.com/cornelk/hashmap v1.0.1
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply go.mod]: done

File: go.sum [hunk 3]
```diff
@@ -53,6 +53,8 @@ github.com/coreos/go-etcd v2.0.0+incompatible/go.mod h1:Jez6KQU2B/sWsbdaef3ED8Nz
 github.com/coreos/go-semver v0.2.0/go.mod h1:nnelYz7RCh+5ahJtPPxZlU+153eP4D4r3EedlOD2RNk=
 github.com/coreos/go-semver v0.3.0 h1:wkHLiw0WNATZnSG7epLsujiMCgPAc9xhjJ4tgnAxmfM=
 github.com/coreos/go-semver v0.3.0/go.mod h1:nnelYz7RCh+5ahJtPPxZlU+153eP4D4r3EedlOD2RNk=
+github.com/cornelk/hashmap v1.0.1 h1:RXGcy29hEdLLV8T6aK4s+BAd4tq4+3Hq50N2GoG0uIg=
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply go.sum]: done

File: node/heartbeat/monitor.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply node/heartbeat/monitor.go]: ok

**Pull request merged**